src/main/java/javaslang/control/Applicative.java

```diff
@@ -0,0 +1,17 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _ _____
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  / /  _  \   Javaslang
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/  \__/  /   Copyright 2014-now Daniel Dietrich
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/___\_/  \_//  \__/_____/    Licensed under the Apache License, Version 2.0
+ */
+package javaslang.control;
+
+public interface Applicative<T> extends Apply<T> {
```

Comment:
This is the shortened version I talked about (got rid of implicit workaround class):

```java
// trait Apply[F[_]] extends Functor[F]
public interface Applicative<F extends Kind<F, ?>, A> extends Functor<A> {

    // def ap[A,B](fa: => F[A])(f: => F[A => B]): F[B]
    <B> Applicative<F, B> ap(Kind<F, ? extends Function<? super A, ? extends B>> f);

    @Override
    <B> Applicative<F, B> map(Function<? super A, ? extends B> f);
}

// type constructor: TYPE<T> is split in TYPE<?> and T
interface Kind<TYPE extends Kind<TYPE, ?>, T> {
}

interface Validation<E, A> extends Kind<Validation<E, ?>, A>, Applicative<Validation<E, ?>, A> {

    // def ap[EE >: E, B](x: => Validation[EE, A => B])(implicit E: Semigroup[EE]): Validation[EE, B]
    <B> Validation<E, B> ap(Validation<E, ? extends Function<? super A, ? extends B>> f);

    @Override
    <B> Validation<E, B> map(Function<? super A, ? extends B> f);

    // def ap[A, B](fa: => Validation[L, A])(f: => Validation[L, A => B]) = fa ap f
    @SuppressWarnings("unchecked")
    @Override
    default <B> Validation<E, B> ap(Kind<Validation<E, ?>, ? extends Function<? super A, ? extends B>> f) {
        return ap((Validation<E, ? extends Function<? super A, ? extends B>>) (Object) f);
    }
}
```